server/sonar-server/src/main/java/org/sonar/server/startup/RenameIssueWidgets.java

```diff
@@ -137,11 +139,18 @@ private WidgetPropertyDto createWidgetProperty(String key, String value, WidgetD
 
   private void updateWidget(DbSession session, WidgetDto widget) {
     dbClient.widgetDao().update(session,
-      widget.setWidgetKey(WIDGET_PROJECT_ISSUE_FILTER)
+      widget.setWidgetKey(getReplacementWidgetKey(session, widget))
         .setUpdatedAt(system.newDate())
         .setConfigured(true));
   }
 
+  private String getReplacementWidgetKey(DbSession session, WidgetDto widget) {
+    DashboardDto dashboard = dbClient.dashboardDao().getNullableByKey(session, widget.getDashboardId());
+    boolean isOnGlobalDashboard = dashboard != null && dashboard.getGlobal();
```

Comment:
You could for instance throw an exception, or ignore it with a warning, or remove the widget.